src/PHPHtmlParser/Selector.php

```diff
@@ -168,7 +169,9 @@ protected function parseSelectorString($selector)
     protected function seek(array $nodes, array $rule, array $options)
     {
         // XPath index
-        if (count($rule['tag']) > 0 &&
+        if ($rule['tag'] instanceof Countable && 
```

Comment:
`$rule['tag']` is compared with a string below, so this check is incorrect. I suggest checking against `!empty` instead of `count`.